server/platform/services/sharedchannel/sync_recv.go

```diff
@@ -79,7 +80,17 @@ func (scs *Service) processSyncMessage(c request.CTX, syncMsg *model.SyncMsg, rc
 		return fmt.Errorf("cannot check channel share state for sync message: %w", err)
 	}
 	if !exists {
-		return fmt.Errorf("cannot process sync message; channel not shared with remote: %w", ErrRemoteIDMismatch)
+		// Update the UI to ensure the link icon is removed
+		scs.notifyClientsForSharedChannelUpdate(targetChannel)
```

Comment:
What's the use case for this?

Reply:
We want to avoid stale UI. With the current flow, it ensures the UI is updated even when a channel is partially unshared (with some remotes but not others - since we call UnshareChannel , which calls the UI update) through unshareChannelIfNoActiveRemotes only). Without this, the UI might not be updated until the channel is completely unshared (all remotes removed).

Reply:
as discussed, I removed the `notifyClientsForSharedChannelUpdate` call. We still need to conclusively validate that the scenario described in the ticket does not happen anymore.